Lib/test/pickletester.py

```diff
@@ -2163,6 +2192,28 @@ def persistent_id(self, obj):
             check({Clearer(): 1, Clearer(): 2})
             check({1: Clearer(), 2: Clearer()})
 
+    @support.cpython_only
+    def test_bad_ext_code(self):
+        # This should never happen in normal circumstances, because the type
+        # and the value of the extesion code is checked in copyreg.add_extension().
+        key = (__name__, 'MyList')
+        def check(code, exc):
+            assert key not in copyreg._extension_registry
+            assert code not in copyreg._inverted_registry
+            with (support.swap_item(copyreg._extension_registry, key, code),
+                  support.swap_item(copyreg._inverted_registry, code, key)):
+                for proto in protocols[2:]:
+                    with self.assertRaises(exc):
+                        self.dumps(MyList, proto)
```

Comment:
```suggestion
					with self.subTest(proto=proto):
                    	with self.assertRaises(exc):
                        	self.dumps(MyList, proto)
```

Just to know which protocol failed (I see tabs and spaces on GH so I hope that if you commit my suggestions, it will be normalized...)

Reply:
This is not important, because the code is the same for all protocols >= 2. Original tests for this feature only use protocol 2.

On other hand, using `subTest()` will produce worse tracebacks if some test fails. Currently the traceback contains the line with the `check()` call. `subTest()` truncates the traceback.